Fabric.Identity.API/scripts/CatalystDosIdentity.psm1

```diff
@@ -528,27 +528,356 @@ function Test-IsClientRegistered {
+#>
+function Get-ApiRegistration {
+	param(
+        [Parameter(Mandatory=$True)] [Uri] $identityUrl,
+        [Parameter(Mandatory=$True)] [string] $apiName,
+        [Parameter(Mandatory=$True)] [string] $accessToken
+    )
+
+	[Uri]$url = "$($identityUrl.OriginalString)/api/apiresource/$apiName"
+
+    $headers = @{"Accept" = "application/json"}
+    if($accessToken){
+        $headers.Add("Authorization", "Bearer $accessToken")
+    }
+
+    $clientResponse = Invoke-RestMethod -Method Get -Uri $url -Headers $headers
+    return $clientResponse
 }
 
-function New-ApiRegistration($identityUrl, $body, $accessToken) {
-    throw [System.NotImplementedException]
+<#
+    .Synopsis
+    Attempts to register an api with fabric identity
+
+    .Description
+    Takes in the identity server url the json request body and an access token.
+
+    .Parameter identityUrl
+    The identity url to post the api registration
+
+    .Parameter body
+    The json request format for creating the api registration
+
+    .Parameter accessToken
+    An access token previously retrieved from the identity server
+
+    .Example
+    New-ApiRegistration -identityUrl "https://server/identity" -body "{"enabled":true, "name": "sample-api", "userClaims":[], "scopes":[]}" -accessToken "eyJhbGciO"
+#>
+function New-ApiRegistration {
+	param(
+        [Parameter(Mandatory=$True)] [Uri] $identityUrl,
+        [Parameter(Mandatory=$True)] [string] $body,
+        [Parameter(Mandatory=$True)] [string] $accessToken
+    )
+
+    $url = "$identityUrl/api/apiresource"
+    $headers = @{"Accept" = "application/json"}
+    if($accessToken){
+        $headers.Add("Authorization", "Bearer $accessToken")
+    }
+
+    try{
+        $registrationResponse = Invoke-RestMethod -Method Post -Uri $url -body $body -ContentType "application/json" -Headers $headers
+        return $registrationResponse.apiSecret
+    }catch{
+        $exception = $_.Exception
+        $apiResourceObject = ConvertFrom-Json -InputObject $body
+        if ((Assert-WebExceptionType -exception $exception -typeCode 409)) {
+            try{
+                Invoke-RestMethod -Method Put -Uri "$url/$($apiResourceObject.name)" -Body $body -ContentType "application/json" -Headers $headers | out-null
+
+                # Reset api secret
+                $apiResponse = Reset-ApiPassword -identityUrl $url -apiName $($apiResourceObject.name) -accessToken $accessToken
+                return $apiResponse.apiSecret
+            }catch{
+                $error = "Unknown error attempting to post api"
+                $exception = $_.Exception
+                if ($null -ne $exception -and $null -ne $exception.Response) {
+                    $error = Get-ErrorFromResponse -response $exception.Response
+                }
+                throw (New-Object -TypeName "System.Net.WebException" "There was an error registering api $($apiResourceObject.name): $error. Halting installation.", $exception)
```

Comment:
This error should be updated to not say it's halting installation

Reply:
yah. That was part of the original script run in the installer.